myst_parser/mathjax.py

```diff
@@ -30,32 +34,34 @@ def override_mathjax(app: Sphinx):
             html_visit_displaymath,  # type: ignore[assignment]
             None,
         )
-    # https://docs.mathjax.org/en/v2.7-latest/options/preprocessors/tex2jax.html#configure-tex2jax
-    if (
-        app.config.mathjax_config is None
-        and app.env.myst_config.update_mathjax  # type: ignore[attr-defined]
-    ):
-        app.config.mathjax_config = {  # type: ignore[attr-defined]
-            "tex2jax": {
-                "inlineMath": [["\\(", "\\)"]],
-                "displayMath": [["\\[", "\\]"]],
-                "processRefs": False,
-                "processEnvironments": False,
-            }
-        }
-    elif app.env.myst_config.update_mathjax:  # type: ignore[attr-defined]
-        if "tex2jax" in app.config.mathjax_config:
+
+    if not app.env.myst_config.update_mathjax:  # type: ignore[attr-defined]
+        return
+
+    mjax_classes = app.env.myst_config.mathjax_classes  # type: ignore[attr-defined]
+
+    if "mathjax3_config" in app.config:
+        # sphinx 4 + mathjax 3
+        app.config.mathjax3_config = app.config.mathjax3_config or {}  # type: ignore[attr-defined]
+        app.config.mathjax3_config.setdefault("options", {})
+        if "processHtmlClass" in app.config.mathjax3_config["options"]:
+            logger.warning(
+                "`mathjax3_config['options']['processHtmlClass']` is being overridden "
+                f"by myst-parser to {mjax_classes}. "
+                "Set `myst_mathjax_classes = None` if this is undesirable."
+            )
+        app.config.mathjax3_config["options"]["processHtmlClass"] = mjax_classes
+    else:
+        # sphinx 3 + mathjax 2
```

Comment:
Should we add "to do" items to deprecate this logic at some point? I dunno how many release cycles we'd want to support but just to make it easier for us to find future code to deprecate and clean up?

Reply:
Nah I thinks ok, since it specifically comments that it is for sphinx 3, and we would only remove if we drop sphinx 3 support